scripts/tarchain.ts

```diff
@@ -0,0 +1,74 @@
+import fs from "fs-extra";
+import path from "path";
+import yargs from "yargs";
+import { spawn, SpawnOptions } from "child_process";
+
+yargs
+  .scriptName("tarchain")
+  .recommendCommands()
+  .demandCommand(1)
+  .strict(true)
+  .showHelpOnFail(true)
+  .command(
+    "run <datadir> <monorepo>",
+    "Create a new tarball that includes the deployed staked CELO contracts",
+    (args) =>
+      args
+        .positional("datadir", { type: "string", description: "Path to devchain data directory" })
+        .positional("monorepo", { type: "string", description: "Path to monorepo" })
+        .option("filename", {
+          type: "string",
+          description: "Filename of output tarball (default: stakedCeloDevchain.tar.gz)",
+        }),
+    (args) =>
+      exitOnError(
+        runCmd(
+          args.datadir!,
+          args.filename != ("" || undefined) ? args.filename : "stakedCeloDevchain.tar.gz",
```

Comment:
Doesn't `("" || undefined)` just always evaluate to `undefined`?

Reply:
It never actually evaluates to `undefined`, as `yargs` parses empty `filename` argument as an empty string, but since its type is `string | undefined` I added an undefined check to make it null safe.